src/terrama2/impl/DataRetrieverHTTP.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+  Copyright (C) 2007 National Institute For Space Research (INPE) - Brazil.
+
+  This file is part of TerraMA2 - a free and open source computational
+  platform for analysis, monitoring, and alert of geo-environmental extremes.
+
+  TerraMA2 is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation, either version 3 of the License,
+  or (at your option) any later version.
+
+  TerraMA2 is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with TerraMA2. See LICENSE. If not, write to
+  TerraMA2 Team at <terrama2-team@dpi.inpe.br>.
+*/
+
+/*!
+  \file terrama2/impl/DataRetrieverHTTP.cpp
+
+  \brief Data Retriever HTTP.
+
+  \author Jean Souza
+*/
+
+// STL
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <functional>
+
+// TerraMA2
+#include "DataRetrieverHTTP.hpp"
+#include "../core/data-model/Filter.hpp"
+#include "../core/utility/Raii.hpp"
+#include "../core/utility/Logger.hpp"
+#include "../core/data-model/Filter.hpp"
+#include "../core/utility/FilterUtils.hpp"
+#include "../core/utility/Utils.hpp"
+
+// TerraLib
+#include <terralib/core/uri/URI.h>
+#include <terralib/core/Exception.h>
+
+// Libcurl
+#include <curl/curl.h>
+
+// Boost
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem/operations.hpp>
+
+// QT
+#include <QObject>
+#include <QDir>
+#include <QDebug>
+#include <QUrl>
+
+terrama2::core::DataRetrieverHTTP::DataRetrieverHTTP(DataProviderPtr dataprovider, std::unique_ptr<CurlWrapperHttp>&& curlwrapper)
+  : DataRetriever(dataprovider),
+    curlwrapper_(std::move(curlwrapper))
+{
+  // Verifies if the HTTP address is valid
+  try
+  {
+    curlwrapper_->verifyURL(dataprovider->uri, dataProvider_->timeout);
+  }
+  catch(const te::Exception& e)
+  {
+    QString errMsg = QObject::tr("HTTP address is invalid! \n\n Details: \n");
+    auto errStr = boost::get_error_info<te::ErrorDescription>(e);
+    if(errStr)
+      errMsg.append(QString::fromStdString(*errStr));
+    errMsg.append(e.what());
+
+    TERRAMA2_LOG_ERROR() << errMsg;
+    throw DataRetrieverException() << ErrorDescription(errMsg);
+  }
+  catch(...)
+  {
+    throw DataRetrieverException() << ErrorDescription(QObject::tr("Unknown Error, HTTP address is invalid!"));
```

Comment:
This is a general catch, don't use specific messages.
This message may be shown to the user.